rpc-server/src/modules/queries/methods.rs

```diff
@@ -241,26 +241,50 @@ async fn view_account(
         account_id,
         block.block_height
     );
-
-    let account = data
-        .db_manager
-        .get_account(account_id, block.block_height)
+    if let Ok(result) = data
```

Comment:
Put a comment explaining the logic of relying on the Err for non-optimistic queries for future us.

Reply:
Or break down so different functions as discussed on the call